File: backend/app/schemas/artifact.py
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class ArtifactCreateInlineRequest(BaseModel):
    """Create an artifact from inline text (Phase 2)."""

    filename: str
    artifact_type: Literal["text", "markdown"]
    text: str = Field(min_length=0)
    title: str | None = None
    mime_type: str | None = None


class ArtifactCreateResponse(BaseModel):
    """Response for `POST /artifacts`."""

    artifact_id: str
    status: Literal["ingested"]
    artifact_type: str
    evidence_unit_count: int


class ArtifactResponse(BaseModel):
    """Artifact metadata response."""

    id: str
    filename: str
    title: str | None
    artifact_type: str
    mime_type: str | None
    storage_uri: str
    checksum: str | None
    size_bytes: int
    created_at: datetime
    updated_at: datetime
    metadata: dict[str, Any] | None = None


class ArtifactListResponse(BaseModel):
    """List artifacts response."""

    items: list[ArtifactResponse]

File: frontend/src/types/artifact.ts
/** Mirrors `ArtifactResponse` / `ArtifactListResponse` from backend `app/schemas/artifact.py`. */

export interface ArtifactResponse {
  id: string;
  filename: string;
  title: string | null;
  artifact_type: string;
  mime_type: string | null;
  storage_uri: string;
  checksum: string | null;
  size_bytes: number;
  created_at: string;
  updated_at: string;
  metadata?: Record<string, unknown> | null;
}

export interface ArtifactListResponse {
  items: ArtifactResponse[];
}
